core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -137,7 +138,7 @@ abstract class AbstractFetcherThread(name: String,
                     // Once we hand off the partition data to the subclass, we can't mess with it any more in this thread
                     processPartitionData(topicAndPartition, currentPartitionFetchState.offset, partitionData)
                   } catch {
-                    case ime: InvalidMessageException =>
+                    case ime: CorruptRecordException =>
```

Comment:
I think my question was not clear. Is `InvalidMessageException` an exception that users of the old clients would catch? If it is, they will have to update their code in the same way as we updated our code here (ie the change breaks compatibility). If this exception is not exposed by the old clients, then it's fine.